tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/TransactionStartController.java

```diff
@@ -76,37 +73,37 @@ protected ModelAndView handleRequestInternal(HttpServletRequest request, HttpSer
 		return result;
 	}
 
-	private ModelAndView startTransaction(Repository repository, HttpServletRequest request,
-			HttpServletResponse response) throws IOException, ClientHTTPException, ServerHTTPException {
-		ProtocolUtil.logRequestParameters(request);
-		Map<String, Object> model = new HashMap<>();
+	@Deprecated
+	ArrayList<TransactionSetting> getIsolationLevel(HttpServletRequest request) {
+		// process legacy isolation level param for backward compatibility with older clients
 
 		ArrayList<TransactionSetting> transactionSettings = new ArrayList<>();
-
-		final IsolationLevel[] isolationLevel = { null };
-
-		// process legacy isolation level param for backward compatibility with older clients
 		final String isolationLevelString = request.getParameter(Protocol.ISOLATION_LEVEL_PARAM_NAME);
 		if (isolationLevelString != null) {
-			final IRI level = Values.iri(isolationLevelString);
 
 			for (IsolationLevel standardLevel : IsolationLevels.values()) {
-				if (standardLevel.getName().equals(level.getLocalName())) {
-					isolationLevel[0] = standardLevel;
+				if (standardLevel.toString().equals(isolationLevelString)) {
```

Comment:
I think this should use `getValue` rather than `toString`.

Reply:
made this change but note isolationlevel.getvalue() is implemented through toString().